x/programs/runtime/config.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package runtime
+
+import "github.com/bytecodealliance/wasmtime-go/v12"
+
+const (
+	defaultMaxWasmStack    = 256 * 1024 * 1024 // 256 MiB
+	defaultWasmThreads     = false
+	defaultFuelMetering    = true
+	defaultWasmMultiMemory = false
+	defaultWasmMemory64    = false
+
+	defaultLimitMaxMemory = 16 * 64 * 1024 // 16 pages
+)
+
+// TODO: review Cranelift fine tune knobs when exposed
+// https://github.com/bytecodealliance/wasmtime-go/pull/188
+
+func NewConfigBuilder(meterMaxUnits uint64) *builder {
+	return &builder{meterMaxUnits: meterMaxUnits}
+}
+
+type builder struct {
```

Comment:
We should also be limiting max depth in the invoke call^

Reply:
plus test adding todo